index.html

```diff
@@ -4545,28 +4545,172 @@ <h2>
       </p>
     </section>
 
-    <section>
-      <h2>
-Key Revocation and Recovery
-      </h2>
 
-      <p>
+
+<section>
+  <h2>
+  Verification Method Rotation
+  </h2>
+  
+  <p>
+    Verification method rotation is a proactive security measure.
+  </p>
+  
+  <p>
+    When a verification method has been active for a long time, 
+    or used for many operations, a controller might wish to perform a rotation.
+  </p>
+
+  <p class="advisement">
+    It is considered a best practice to perform verification method rotation on a regular basis.
+  </p>
+  
+  <p>
+    Section <a href="#method-operations"></a> specifies the <a>DID</a> operations to
+    be supported by a <a>DID method</a> specification, 
+    including <a href="#update">update</a>
+    which MUST be used to perform a verification method rotation.
+  </p>
+  
+  <p>
+    A controller performs a rotation when they change 
+    <code>publicKeyJwk</code> or <code>publicKeyBase58</code>, or other verification material
+    without changing the <code>id</code> of the verification method.
+  </p>
+  
+  <p>
+    Unlike revocation, rotation allows for new verifications associated with the <code>id</code>, 
+    to be valid, while invalidating old verifications that were produced with the older material.
+  </p>
+  
+  <p>
+    Rotation is a useful mechanism for protecting against verification method compromise, 
+    since frequent rotation of a verification method by the controller, 
+    reduces the value of a compromised verification method to an attacker.
+  </p>
+  
+  <p class="note">
+    Higher security environments tend to employ more frequent verification method rotation.
+  </p>
+  
+  <p class="note">
+    Frequent rotation of a verification method might be frustrating 
+    for parties that are forced to continiously renew or refresh associated credentials.
+  </p>
+  
+</section>
+
+<section>
+<h2>
+Verification Method Revocation
+</h2>
+
+<p>
+  Verification method revocation is a reactive security measure.
+</p>
+
+<p>
+  If a verification method is no longer exclusivley accessible to the 
+  controller or parties trusted to act on behalf of the controller, 
+  it should be revoked immediatly to reduce the risk of masquerading, theft and fraud.
+</p>
+
+<p class="advisement">
+  It is considered a best practice to support key revocation.
+</p>
+
+<p class="advisement">
+A controller should revoke any verification method that is believed to be compromised immediatly.
+</p>
+
+<p class="advisement">
+Cryptographic verifications associated with revoked verification methods should be considered invalid  
```

Comment:
I could possibly live with the following rewording:

Cryptographic verifications associated with a revoked verification method are not valid. However, knowing whether a verification was made with a revoked verification method is trickier than it may seem.

Some DID methods provide the ability to look back at the state of a DID at a point in time, or at a particular version of the DID document. When such a feature is combined with a reliable way to determin the time or DID version that existed when a cryptographically verifiable statement was made, then revocation does not undo that statement. This can be the basis for using DIDs to make binding commitments (e.g., to sign a mortgage). If these conditions are met, revocation is not retroactive; it only nullifies future use of the method.

But in order for such semantics to be safe, the second condition -- an ability to know what the state of the DID document was at the time the assertion was made -- MUST apply. Without that guarantee, someone could discover a revoked key and use it to make cryptographically verifiable statements with a simulated date in the past.

Some DID methods only allow the retrieval of the current state of a DID. When this is true, or when the state of a DID at the time of cryptographically verifiable statement cannot be reliably determined, then the only safe interpretation of revocation is to make it apply both forward and backward in time. DID ecosystems that take this approach essentially provide cryptographically verifiable statements as ephemeral tokens that can be invalidated at any time by the DID controller.

Reply:
@dhh1128, I agree with your reword. I think it's a little strange to talk about the "state of a DID" (which is just an identifier), so we should perhaps instead talk about DID Documents or resolution. That's just an editorial quibble though. A slightly more important suggestion, I think, would be such that we don't so strongly state that if a DID method only allows retrieval of the current state of a DID that there are no alternative solutions to preventing undesirable "retroactive revocation". We shouldn't be so prescriptive -- instead of saying:

> Some DID methods only allow the retrieval of the current state of a DID.

We can say:

> In some cases, it may only be possible to obtain a current DID Document.

That way, future clever solutions that layer on top of DID methods that do not support versioning directly are not needlessly discouraged.

Reply:
Good point about "state of a DID." I once poked a colleague about that very error, so I'm being inconsistent to get sloppy now. :-)

I agree that the rewording you suggested improves the text.

Reply:
I don't believe the proposed language is an improvement.

However, I hope I have captured its spirit in my latest change set, by stating that revocation and rotation only apply to the latest version of a did document.... and clarifying how revoked material is commonly treated by other software systems.